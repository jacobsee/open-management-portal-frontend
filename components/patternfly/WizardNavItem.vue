<template>
  <li class="pf-c-wizard__nav-item">
    <a
      class="pf-c-wizard__nav-link"
      v-bind:class="{ 'pf-m-current': isCurrent }"
      @click="$emit('navigate', item.key)"
    >
      <slot />
    </a>
  </li>
</template>

<script>
export default {
  props: ['item', 'isCurrent']
}
</script>
