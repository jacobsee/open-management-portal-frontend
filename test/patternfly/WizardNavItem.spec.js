import { mount } from '@vue/test-utils'
import WizardNavItem from '@/components/patternfly/WizardNavItem.vue'

const testProps = {
  isCurrent: true,
  item: {
    key: 'first-form'
  }
}

describe('WizardNavItem', () => {
  test('is a Vue instance', () => {
    const wrapper = mount(WizardNavItem, {
      propsData: testProps
    })
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('renders properly', () => {
    const wrapper = mount(WizardNavItem, {
      propsData: testProps
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
