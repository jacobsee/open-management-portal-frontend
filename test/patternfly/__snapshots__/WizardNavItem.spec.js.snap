// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WizardNavItem renders properly 1`] = `"<li class=\\"pf-c-wizard__nav-item\\"><a class=\\"pf-c-wizard__nav-link pf-m-current\\"></a></li>"`;
